src/main/java/com/epam/training/array/ArrayLogic.java

```diff
@@ -1,16 +1,72 @@
 package com.epam.training.array;
 
+import java.util.Arrays;
+import java.util.Collections;
+
 public class ArrayLogic {
 
     //TODO: write it using binary search
-    public boolean exists(Array array) {
-        throw new UnsupportedOperationException();
+    public boolean exists(int[] array, int searchedNumber) {
+        sort(array);
+        int searchResult = Arrays.binarySearch(array, searchedNumber);
+
+        int arrayLength = array.length;
+        int indexOfNumber = 0;
+
+        for (int i = 0; i < arrayLength; i++) {
+            if (array[i] == searchedNumber) {
+                indexOfNumber =  i;
+            }
+        }
+
+        if (searchResult == indexOfNumber){
+            return true;
+        }else {
+            return false;
+        }
+    }
+
+    public int[] findPrimeNumbers(int[] array) {
+        int j = 0;
+        int k = 0;
+        int l = 0;
+        int[] primeNumbers = new int[array.length];
+
+        for (int i = 0; i < array.length; i++) {
+            j = 2;
+            k = 1;
+            while (j < array[i]) {
+                if (array[i] % j == 0) {
+                    k = 0;
+                    break;
+                }
+                j++;
+            }
+            if (k == 1) {
+                primeNumbers[l] = array[i];
+                l++;
+            }
+        }
+        return primeNumbers;
     }
 
-    public Array findPrimeNumbers(Array array) {
-        throw new UnsupportedOperationException();
+    //TODO: add other methods here
+
+    public int[] sort(int[] array) {
+        Arrays.sort(array);
+        return array;
```

Comment:
what do you want to do modifier it by link or cope then sort and return result, you can't mix it